runner/execer/os/os_execer.go

```diff
@@ -193,16 +194,21 @@ func (e *osExecer) monitorMem(p *osProcess) {
 						"jobID":       p.JobID,
 						"taskID":      p.TaskID,
 					}).Infof("Increased mem_cap utilization for pid %d to %d", pid, int(memUsagePct*100))
-				ps, err := exec.Command("ps", "-u", os.Getenv("USER"), "-opid,sess,ppid,pgid,rss,args").CombinedOutput()
+
+				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```

Comment:
Quick nit - could you add a comment providing context for the... context. It isn't immediately clear looking at this why exec.Cmd.CombinedOutput() would need a timeout